src/os_common/os_alloc.c

```diff
@@ -139,7 +139,7 @@ __realloc_func(WT_SESSION_IMPL *session, size_t *bytes_allocated_ret, size_t byt
      */
     tmpp = p;
     if (session != NULL && FLD_ISSET(S2C(session)->debug_flags, WT_CONN_DEBUG_REALLOC_MALLOC) &&
-      (bytes_allocated_ret != NULL)) {
```

Comment:
The previous check i.e. `bytes_allocated_ret` is almost never NULL. I think the reason that the check existed in the first place was to avoid performing a `memcpy`, however, it doesn't seem to accomplish the same. 

As per my understanding of this code, I think this is the correct change as if `tmpp` is NULL there is no need to perform a `memcpy`.

Reply:
@Siddhartha8899, @lting0804, the reason that `bytes_allocated_ret` is checked against NULL is that in the subordinate clause, `memcpy(p, tmpp, *bytes_allocated_ret);` is done, and that would give an access violation (if it were NULL).  Whether or not it's almost never NULL, I think we need to handle the case.  So I think you still need `bytes_allocated_ret != NULL` as part of the check.

Granted, it's a weird case, but I think this would crash:
```
void *mem = malloc(10);
__realloc_func(session, NULL, 50, false, &mem);
```

Reply:
ahh yes, thanks @ddanderson, that makes sense.

Reply:
@Siddhartha8899, I think it's okay to use `bytes_allocated` instead of `*bytes_allocated_ret` in the `memcpy`.

Reply:
Sorry if I am misunderstanding this, however, if we use `bytes_allocated` instead of `*bytes_allocated_ret`, then there is no need to check if bytes_allocated_ret is a non-null value, right?

Reply:
Ah, yes, got it.  You're right - change the usage at the memcpy, then you don't need the `bytes_allocate_ret` check at line 141.  Can either @lting0804 or @Siddhartha8899 push this change so we can review it?

Reply:
Hi @ddanderson @Siddhartha8899, 

I've reverted it to include the original null check for `bytes_allocated_ret` and moved the tmpp null check before memcpy :) thanks @mm-ng